### PR TITLE
Added Youtube support

### LINKED ISSUE #290: Add Youtube support
**I'm submitting a ...**
- [ ] bug report
- [x] feature request

**Current behavior:**

**Expected behavior:**

**Steps to reproduce:**

**Yes, I would like to work on this issue**

### PR DESCRIPTION
**Pull request by bhaveshAn**
Fixes #290 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `master` branch.
- [x] I have added necessary documentation (if appropriate)

#### Changes proposed in this pull request:

- Added Youtube support

@gabru-md @cclauss @mariobehling please review. Providing the heroku deployment at https://immense-anchorage-24063.herokuapp.com/
Thanks !!

### CODE REVIEW

**Comment by gabru-md** [comment]
@bhaveshAn great work.
I would like to bring it to your notice @mariobehling that we must start searching using the fullname of the scrapper as @cclauss suggests so that we donot face problems like bhavesh had i.e. using `tyoutube` instead of `youtube`.
@cclauss please comment. Rest bhavesh did good job

File: app/scrapers/__init__.py [hunk 1]
```diff
@@ -34,7 +36,7 @@ def small_test():
 
 
 def feedgen(query, engine, count=10):
-    if engine == 'q':
+    if engine in ['q', 't']:
```
Comment (cclauss) [inline app/scrapers/__init__.py]: Putting this logic in two different files can be the source of future bugs.  Please encapsulate this logic in one file or the other but not in both.

File: app/server.py [hunk 2]
```diff
@@ -49,7 +49,7 @@ def search(search_engine):
             err = [400, 'Not Found - missing query', qformat]
             return bad_request(err)
 
-        if engine[0] == 'q':
+        if engine[0] in ['q', 't']:
```
Comment (cclauss) [inline app/server.py]: Putting this logic in two different files can be the source of future bugs.  Please encapsulate this logic in one file or the other but not in both.

File: app/scrapers/youtube.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+class Youtube(Scraper):
+    """Scraper class for Youtube"""
+
+    def __init__(self):
+        self.url = 'https://www.youtube.com/results'
+        self.queryKey = 'search_query'
+
+    def parseResponse(self, soup):
+        """ Parse the response and return set of urls
```
Comment (cclauss) [inline app/scrapers/youtube.py]: This is not a __set of URLs__.  It is a __list of dicts__.  A set guarantees not repeats and this code does not do that.  The items in the list are dicts that contain titles and links (urls).

[[Tile1,url1], [Title2, url2],..] should be changed to [{'title': Tile1, 'link': url1}, {'title': Tile2, 'link': url2}, ...] to make the result more clear to the reader

File: app/scrapers/__init__.py (hunk 1 above)
Comment (bhaveshAn) [reply app/scrapers/__init__.py]: @cclauss Please review again

File: app/scrapers/youtube.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    """Scraper class for Youtube"""
+
+    def __init__(self):
+        self.url = 'https://www.youtube.com/results'
+        self.queryKey = 'search_query'
+
+    def parseResponse(self, soup):
+        """ Parse the response and return list of urls
+        Returns: urls (list)
+                [[Tile1,url1], [Title2, url2],..]
```
Comment (cclauss) [inline app/scrapers/youtube.py]: The returned values are not urls. They are dicts.

I know these changes were not tested under Python 3.

**Comment by bhaveshAn** [comment]
@mariobehling @niranjan94 Please review.

**Comment by mariobehling** [comment]
Thanks!

**Pull request merged**